test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -747,6 +755,83 @@ def __are_requirements_matching(param_dic=None, test=None):
             if not eff_tc_req['no_comm_on_mom']:
                 return False
 
+    def hardware_report(self):
```

Comment:
how about renaming this to `check_and_report_hardware_status` to make more readable?